identity_did/src/verification/method_type.rs

```diff
@@ -37,10 +44,32 @@ impl FromStr for MethodType {
 
   fn from_str(string: &str) -> Result<Self, Self::Err> {
     match string {
-      "Ed25519VerificationKey2018" => Ok(Self::Ed25519VerificationKey2018),
-      "X25519KeyAgreementKey2019" => Ok(Self::X25519KeyAgreementKey2019),
-      // other => Ok(Self::Other(other.to_owned())),
-      _ => Err(Error::UnknownMethodType),
+      "Ed25519VerificationKey2018" => Ok(Self::ED25519_VERIFICATION_KEY_2018),
+      "X25519KeyAgreementKey2019" => Ok(Self::X25519_KEY_AGREEMENT_KEY_2019),
+      _ => Ok(Self(Cow::Owned(string.to_owned()))),
+    }
+  }
+}
+
+//TODO: Also implement From<String>?
+
+// TODO: Is this the right place for this? Is this even needed?
```

Comment:
Probably not. Ideally this would live with that specific `KeyType` implementation, but since that is in `identity_core` we can perhaps just move it to a separate module?

Reply:
I will merge this now so we can quickly iterate, but we should definitely come back to this.